### ISSUE TITLE
Installation error

### ISSUE THREAD

**Issue by saimn** [pytroll/aggdraw#31]
Since `freetype-config` is not available on Archlinux the installation fails (actually the update to the latest version fails, so something changed there):
```
❯ LANG=C pip install aggdraw
Collecting aggdraw
  Using cached https://files.pythonhosted.org/packages/49/e8/5021fba0bf22434c1f15700f01fb67750a86bae0f320d39aee53a61ed968/aggdraw-1.3.5.tar.gz
Building wheels for collected packages: aggdraw
  Running setup.py bdist_wheel for aggdraw ... error
  Complete output from command /home/simon/.pyenv/versions/3.7.0/bin/python3.7 -u -c "import setuptools, tokenize;__file__='/tmp/simon/pip-install-yez_bvvk/aggdraw/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/simon/pip-wheel-1qocje58 --python-tag cp37:
  Using ctypes to find freetype library...
  === freetype found: '/tmp/simon/pip-install-yez_bvvk'
  running bdist_wheel
  running build
  running build_ext
  building 'aggdraw' extension
  creating build
  creating build/temp.linux-x86_64-3.7
  creating build/temp.linux-x86_64-3.7/agg2
  creating build/temp.linux-x86_64-3.7/agg2/src
  creating build/temp.linux-x86_64-3.7/agg2/font_freetype
  ccache /usr/local/bin/cc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -O3 -Wall -fPIC -DVERSION=1.3.5 -DHAVE_FREETYPE2 -Iagg2/include -Iagg2/font_freetype -I/tmp/simon/pip-install-yez_bvvk/include -I/tmp/simon/pip-install-yez_bvvk/include/freetype -I/tmp/simon/pip-install-yez_bvvk/include/freetype2 -I/home/simon/.pyenv/versions/3.7.0/include/python3.7m -c aggdraw.cxx -o build/temp.linux-x86_64-3.7/aggdraw.o
  In file included from aggdraw.cxx:84:
  agg2/font_freetype/agg_font_freetype.h:23:10: fatal error: ft2build.h: No such file or directory
   #include <ft2build.h>
            ^~~~~~~~~~~~
  compilation terminated.
  error: command 'ccache' failed with exit status 1
```

It could be related to #30, as `ctypes.util.find_library` does not return the full path of the so file (in my case it gives `'libfreetype.so.6'`):
https://github.com/pytroll/aggdraw/blob/fb26dedd70864e19f117dc24b1528328dc6d30b4/setup.py#L39-L40
So the flags are added but the paths are not valid.

**Comment by djhoese** [comment]
Thanks for the issue. Obviously I didn't realize that linux behaved differently. I did find this python bug report which addresses the issue but they couldn't make in all cases: https://bugs.python.org/issue21042

I think the obvious fix to make it at least installable would be to check if the library path actually exists and if it doesn't, don't try to use freetype. I should also probably add an environment variable override `os.getenv(...)`.

The next solution would be if the library doesn't exist and we're on linux, try to find it using the python prefix:

```python
os.path.isfile(os.path.join(sys.prefix, 'lib', find_library('freetype')))
```

I'll try to apply some of these later today.

**Comment by saimn** [comment]
Yes, at least a fix to allow the installation would be nice, even without freetype.
Then another solution could be to use pkgconfig when `freetype-config` is not available (with or without its Python wrapper):
```
In [1]: import pkgconfig

In [2]: pkgconfig.exists('freetype')
Out[2]: False

In [3]: pkgconfig.exists('freetype2')
Out[3]: True

In [4]: pkgconfig.variables('freetype2')
Out[4]: 
{'includedir': '/usr/include',
 'libdir': '/usr/lib',
 'exec_prefix': '/usr',
 'prefix': '/usr',
 'pcfiledir': '/usr/lib/pkgconfig'}
```

**Comment by djhoese** [comment]
Oh very interesting. The only issue I have with that is that `pkgconfig` is not in the standard library.

**Comment by djhoese** [comment]
@saimn I made some changes to the setup.py. Before I make a release could you try installing it (now merged to the master branch of this repository). You can do so using `pip install git+https://github.com/pytroll/aggdraw.git` and it *should* print out if freetype was found.

See https://github.com/pytroll/aggdraw/pull/32 for more info on what I changed.

**Comment by saimn** [comment]
@djhoese - Thanks, though it does not work:
```❯ LANG=C pip install git+https://github.com/pytroll/aggdraw.git
Collecting git+https://github.com/pytroll/aggdraw.git
  Cloning https://github.com/pytroll/aggdraw.git to /tmp/simon/pip-req-build-kcp2lbq2
Building wheels for collected packages: aggdraw
  Running setup.py bdist_wheel for aggdraw ... error
  Complete output from command /home/simon/.pyenv/versions/3.7.0/bin/python3.7 -u -c "import setuptools, tokenize;__file__='/tmp/simon/pip-req-build-kcp2lbq2/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/simon/pip-wheel-spp743c1 --python-tag cp37:
  Trying freetype-config to find freetype library...
  Using ctypes to find freetype library...
  === freetype found: '/tmp/simon'
  running bdist_wheel
  running build
  running build_ext
  building 'aggdraw' extension
  creating build
  creating build/temp.linux-x86_64-3.7
  creating build/temp.linux-x86_64-3.7/agg2
  creating build/temp.linux-x86_64-3.7/agg2/src
  creating build/temp.linux-x86_64-3.7/agg2/font_freetype
  ccache /usr/local/bin/cc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -O3 -Wall -fPIC -DVERSION=1.3.6a0.dev0 -DHAVE_FREETYPE2 -Iagg2/include -Iagg2/font_freetype -I/tmp/simon/include -I/tmp/simon/include/freetype -I/tmp/simon/include/freetype2 -I/home/simon/.pyenv/versions/3.7.0/include/python3.7m -c aggdraw.cxx -o build/temp.linux-x86_64-3.7/aggdraw.o
  In file included from aggdraw.cxx:84:
  agg2/font_freetype/agg_font_freetype.h:23:10: fatal error: ft2build.h: No such file or directory
   #include <ft2build.h>
            ^~~~~~~~~~~~
  compilation terminated.
  error: command 'ccache' failed with exit status 1
```
I will give more detail in #32.

**Comment by djhoese** [comment]
I made #33 you could install that branch specifically by adding a `@bugfix-freetype-config` to your pip install command and see what you get.

**Comment by saimn** [comment]
@djhoese - Thanks, it works:
```
    Running command python setup.py egg_info
    Trying freetype-config to find freetype library...
    Using ctypes to find freetype library...
    === freetype found: '/usr'
```

**Comment by djhoese** [comment]
Thanks for testing this @saimn. This has now been released on PyPI as aggdraw 0.3.6. A conda-forge package is also being prepared: https://github.com/conda-forge/aggdraw-feedstock/pull/3#issuecomment-418467717

**Comment by saimn** [comment]
Thanks for the quick fix :slightly_smiling_face:
